torch/_inductor/ir.py

```diff
@@ -1989,12 +1997,16 @@ def unwrap_views(target):
                 return unwrap_views(target.data)
             return target
 
-        return unwrap_views(self.target).layout
+        result = unwrap_views(self.target)
+        assert isinstance(result, Buffer), "MutationLayout must refer to a buffer"
+        return result
```

Comment:
I was seeing accuracy failures and narrowed it down to certain fallback kernels like scatter fallback not calling `realize_users_of`, so this strictly enforces that creating a `MutationLayout` will mark the buffer as mutated.